moveit_core/robot_state/test/test_aabb.cpp

```diff
@@ -0,0 +1,274 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2008, Willow Garage, Inc.
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Willow Garage nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/** \author Martin Pecka*/
+
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit_resources/config.h>
+#include <urdf_parser/urdf_parser.h>
+#include <fstream>
+#include <boost/filesystem.hpp>
+#include <gtest/gtest.h>
+
+class TestAABB : public testing::Test
+{
+protected:
+  std::string readFileToString(boost::filesystem::path path) const
+  {
+    std::string file_string;
+    std::fstream file(path.string().c_str(), std::fstream::in);
+    if (file.is_open())
+    {
+      std::string line;
+      while (file.good())
+      {
+        std::getline(file, line);
+        file_string += (line + "\n");
+      }
+      file.close();
+    }
+    return file_string;
+  }
+
+  virtual void SetUp(){};
+
+  robot_state::RobotState loadModel(const std::string urdf, const std::string srdf)
+  {
+    boost::shared_ptr<urdf::ModelInterface> parsed_urdf(urdf::parseURDF(urdf));
```

Comment:
i know we have mixed API, but is it possible to use std::shared_ptr here instead?

Reply:
This is targeted at indigo-devel. Wouldn't that be a problem? The system gcc on Trusty doesn't know about C++11 (I think).

Reply:
Ah yes, it should be boost if you are on indigo. Preferably this would only target kinetic, but I understand you might have constraints for this.  When we cherry-pick this to kinetic we'll have to switch to std::shared_ptr